contributors/design-proposals/local-pv-provisioner.md

```diff
@@ -0,0 +1,156 @@
+# Local PV Distributed Static Provisioner
+Authors: @verult, @msau42
+
+The current preference is [design 3](#design-3-event-driven-with-custom-resources).
+        
+## Design Goals
+1. Limit all PV object interactions to a single master pod in order to minimize node access to the Kubernetes system. The original discussion is [here](https://github.com/kubernetes/community/pull/989#discussion_r135397619).
+1. Assume disk deletion is asynchronous, as required by block volumes.
+
+## Related Works
+* Original provisioner design, [here](https://github.com/kubernetes-incubator/external-storage/tree/master/local-volume/provisioner#design).
+* Disk cleaner (deleter) implementation for block volumes, [here](https://github.com/kubernetes-incubator/external-storage/pull/312).
+
+## Design 1: Master-Heavy
+
+### High Level Overview
+The static provisioner will be split into a single master (ideally placed on the master node) and many minions (one per node). The master contains the bulk of provisioner logic, including syncing PV objects with the API server and triggering discovery and deletion. Minions are left to perform the physical discovery and deletion of the local volumes on its node. gRPC is used to communicate between master and minions, with the master as the client and each minion as a separate server.
+
+The master keeps track of `MinionInfo`, a collection containing one entry for each minion keeping track of relevant information. The master also runs a group of goroutines, one for each minion, which controls and interacts with the minions.
+
+### Minion Provisioner
+The minion provisioner exposes the following RPCs:
+
+| RPC                           | Description |
+| --- | --- |
+| `GetMinionName()`             | Returns the minion provisioner's name. |
+| `GetMinionStatus()`           | Returns the minion status, as described in the next section. |
+| `Delete(PersistentVolume)`    | Starts the asynchronous delete of the given PV. |
+
+In addition, the Discoverer and Deleter from the previous design are kept here, but they only perform filesystem operations. All other logic is moved to the master.
+
+### MinionStatus
+The following fields are retrieved as part of the `GetMinionStatus()` call:
+
+| Field               | Description |
+| --- | --- |
+| `LocalPVs`          | A collection of PVs mounted in the local storage directory on the node. The discoverer is triggered to acquire this information. |
+| `DeletesInProgress` | A collection of PVs currently being deleted asynchronously. This status is managed by the cleaner. |
```

Comment:
Can we merge `DeletesInProgress` with the `Volume` information?  Maybe just have a `Status` field in the `Volume` with some values like: `Available, DeleteInProgress, DeleteError`, etc.